src/QuadVelocityController.cpp

```diff
@@ -31,27 +33,33 @@ void QuadVelocityController::setTargetVelocity(geometry_msgs::Twist twist)
 {
     throttle_pid_.setSetpoint(twist.linear.z);
     pitch_pid_.setSetpoint(twist.linear.x);
-    roll_pid_.setSetpoint(twist.linear.y);
+    roll_pid_.setSetpoint(-twist.linear.y);
     yaw_pid_.setSetpoint(twist.angular.z);
 }
 
 // Needs to be called at regular intervals in order to keep catching the latest velocities.
-iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update()
+iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update(const ros::Time& time)
 {
-    // Get the time delta
     geometry_msgs::Vector3 velocities;
     getVelocities(velocities);
-    double time_delta = 0.0;
+
+    // Get the time delta
+    if (last_time_ == ros::Time(0)) {
+        last_time_ = time;
```

Comment:
I can't think of a use case where we need a PID controller to not implement the full PID model its designed for on every iteration. Its also opening us up for accidental behaviour if its not explicitly called out in the PID class.

In most applications you'll never notice that only the p term is returned on the first output.